hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -323,32 +322,60 @@ private object HoodieMergeOnReadRDD {
 
   def scanLog(split: HoodieMergeOnReadFileSplit, logSchema: Schema, config: Configuration): HoodieMergedLogRecordScanner = {
     val fs = FSUtils.getFs(split.tablePath, config)
-    val partitionPath: String = if (split.logPaths.isEmpty || split.logPaths.get.asJava.isEmpty) {
-      null
+    val logFiles = split.logFiles.get
+
+    if (HoodieTableMetadata.isMetadataTable(split.tablePath)) {
```

Comment:
I need some clarification here.  So why is metadata table treated differently here?  It's basically another Hudi table.

Reply:
It's, but it's actually using its own standalone scanner (meaning it couldn't be read with the default one).

I agree that we should trim down as much as possible such bifurcations and instead make sure that existing Hudi components are extended in a way that can handle MT, without a need for specialized one

Reply:
I see.  That makes sense.  Yeah, at some point we should see how to reduce such bifurcations and make them easier to maintain.